server/src/test/java/io/littlehorse/common/model/metadatacommand/PrincipalAdministrationTest.java

```diff
@@ -270,12 +271,76 @@ void shouldNotAllowPrincipalWithoutGlobalACLsToHaveAnACLThatPointsToTenantResour
                 .isNotNull()
                 .isInstanceOf(LHApiException.class)
                 .hasMessage(
-                        "INVALID_ARGUMENT: PutPrincipalRequest does not allow non-Admin users to have any permissions on tenants");
+                        "INVALID_ARGUMENT: PutPrincipalRequest does not allow Per-Tenant ACLs containing permissions over Tenants or Principals.");
+    }
+
+    @Test
+    void shouldNotAllowPrincipalToHaveAPerTenantACLThatPointsToPrincipalResource() {
+        // Principal with global write permissions over principal metadata tries to create principal with perTenantAcl
+        // of Principal
+        putPrincipalRequest.setPerTenantAcls(Map.of(tenantId, TestUtil.singleAclWithPrincipalResource()));
+        MetadataCommandModel command = sendCommand(putPrincipalRequest);
+
+        ArgumentCaptor<Exception> exceptionArgumentCaptor = ArgumentCaptor.forClass(Exception.class);
+
+        verify(server).sendErrorToClient(eq(command.getCommandId()), exceptionArgumentCaptor.capture());
+
+        Exception thrown = exceptionArgumentCaptor.getValue();
+        assertThat(thrown)
+                .isNotNull()
+                .isInstanceOf(LHApiException.class)
+                .hasMessage(
+                        "INVALID_ARGUMENT: PutPrincipalRequest does not allow Per-Tenant ACLs containing permissions over Tenants or Principals.");
+    }
+
+    @Test
+    public void shouldNotAllowPrincipalWithAllResourcePerTenantACLToAccessPrincipalResource() {
+        StoredGetable storedRequester =
+                defaultStore.get(new PrincipalIdModel(requesterId).getStoreableKey(), StoredGetable.class);
+        PrincipalModel requester = (PrincipalModel) storedRequester.getStoredObject();
+        requester.getPerTenantAcls().put(tenantId, TestUtil.singleAdminAcl("test"));
+        requester.setGlobalAcls(null);
+        defaultStore.put(new StoredGetable<>(requester));
+
+        MetadataCommandModel command = sendCommand(putPrincipalRequest);
+
+        ArgumentCaptor<Exception> exceptionArgumentCaptor = ArgumentCaptor.forClass(Exception.class);
+
+        verify(server).sendErrorToClient(eq(command.getCommandId()), exceptionArgumentCaptor.capture());
+
+        Exception thrown = exceptionArgumentCaptor.getValue();
+        assertThat(thrown)
+                .isNotNull()
+                .isInstanceOf(LHApiException.class)
+                .hasMessage("PERMISSION_DENIED: Missing permission WRITE_METADATA over resource ACL_PRINCIPAL.");
+    }
+
+    @Test
+    public void shouldNotAllowPrincipalWithAllResourcePerTenantACLToAccessTenantResource() {
+        StoredGetable storedRequester =
```

Comment:
Avoid using production code in tests, later it is needed to mantain both places and maybe we can be lost on what would be a real problem, something that can be done is setup an integration test to see the real behavior under certain conditions.